Add quick start section and gif demo to accompany it.

madhu-GG opened the pull request:
Make sure you have checked all steps below.

### Prerequisite
* [ ] Please consider implementing the feature as a hook script or plugin as a first step.
  * pyenv has some powerful support for plugins and hook scripts. Please refer to [Authoring plugins](https://github.com/pyenv/pyenv/wiki/Authoring-plugins) for details and try to implement it as a plugin if possible.
* [ ] Please consider contributing the patch upstream to [rbenv](https://github.com/rbenv/rbenv), since we have borrowed most of the code from that project.
  * We occasionally import the changes from rbenv. In general, you can expect changes made in rbenv will be imported to pyenv too, eventually.
  * Generally speaking, we prefer not to make changes in the core in order to keep compatibility with rbenv.
* [ ] My PR addresses the following pyenv issue (if any)
  - Closes https://github.com/pyenv/pyenv/issues/3033

### Description
- [ ] Here are some details about my PR

This is a PR to add a quick start section to the github README.md along with a quick demo to show how to install and switch to a different python version using pyenv.

### Tests
- [ ] My PR adds the following unit tests (if any)

Not Applicable.

native-api reviewed (commented):
This essentially duplicates the installation and usage section -- and not in much fewer words, either!

Maybe rather move the installation section higher and/or make it more concise?

The gif is a good idea. Maybe worth showcasing the prefix resolution in it?
If the basic basic scenario is "install the latest CPython release", it could rather be done with `pyenv install 3`.

madhu-GG commented:
> This essentially duplicates the installation and usage sections -- and not in much fewer words, either!
> 
> Maybe rather move the installation section higher and/or make it more concise?
> 
> The gif is a good idea. Maybe worth showcasing the prefix resolution in it? If the basic basic scenario is "install the latest CPython release", it could rather be done with `pyenv install 3`.

1. Yes, looking back at the "Quick Start" it does stretch too much to be quick start. I will try to move the installation section and make it more concise.

2. I admit I don't know what prefix resolution is. Which makes showing it nifty.

native-api commented:
FYI here's the last discussion which raised similar concerns: https://github.com/pyenv/pyenv/issues/1677

madhu-GG commented:
@native-api Hey, I have incorporated the changes you mentioned. Please let me know if it requires further changes.

Changes brief:

1. I updated the GIF to showcase the prefix resolution and moved it inside the Usage section.
2. I have removed the 'Quick Start' section that I introduced, and instead moved the existing installation and usage sections towards the top just after the Table of Contents.
3. The 'How it works' and 'Advanced configuration' have been moved to the bottom just before 'Contributing' section.
4. Within the Installation and Usage sections
  - I have collapsed the details for each platform + shell. This way, the user only needs to expand the sections relevant to their platform and shell.
  - I have also updated numbering and added horizontal divider lines to make these two sections more readable.

native-api commented:
There are conflicts with `master`

madhu-GG commented:
> There are conflicts with `master`

Yeah I will fix those and push my changes.

madhu-GG commented:
> There are conflicts with `master`

resolved.

native-api reviewed (approved):
Okay, these are... pretty compact!
The interactive elements may confuse someone, but I think we can give this a whirl!
I hope you don't mind being CC'ed for any relevant feedback from users?

native-api commented on README.md:
```diff
@@ -278,117 +166,129 @@ so you won't get Windows-specific functionality.
+
+**if you have just upgraded to pyenv version >= 2.3.0**
+
+The startup logic and instructions have been updated for simplicity in 2.3.0.
+The previous, more complicated configuration scheme for 2.0.0-2.2.5 still works.
+
+* Define environment variable `PYENV_ROOT` to point to the path where
+  Pyenv will store its data. `$HOME/.pyenv` is the default.
+  If you installed Pyenv via Git checkout, we recommend
+  to set it to the same location as where you cloned it.
+* Add the `pyenv` executable to your `PATH` if it's not already there
```
IMO This can be compacted, too (and probably deleted altogether after some time). Currently, it takes disproportionate space and visual weight for how situational it is.

native-api commented:
Thank you very much for your contribution! :tada:

madhu-GG commented:
> Okay, these are... pretty compact! The interactive elements may confuse someone, but I think we can give this a whirl! I hope you don't mind being CC'ed for any relevant feedback from users?

Sorry for the late reply. Sure I will be happy to reply. Although I guess I have to come back here more often.

madhu-GG commented:
> Thank you very much for your contribution! 🎉

Thanks for the encouragement. I do want to continue contributing, even if very seasonally.
